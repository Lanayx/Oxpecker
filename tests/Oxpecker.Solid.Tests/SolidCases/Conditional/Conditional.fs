module Oxpecker.Solid.Tests.Cases.Conditional

open Oxpecker.Solid

[<SolidComponent>]
let Test (show: bool) =
    let x, _ = createSignal 7
    div() {
        Show(when'=show, fallback = div() { "World" }) {
            div() {
                "Hello"
            }
        }
        Switch(fallback= p(){ x(); "is between 5 and 10" }) {
            Match(when'= (x() > 10)) {
                p() { x(); "is greater than 10" }
            }
            Match(when'= (x() < 5)) {
                p() { x(); "is less than 5" }
            }
        }
    }
