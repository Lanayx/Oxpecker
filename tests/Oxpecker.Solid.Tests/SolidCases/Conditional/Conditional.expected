import { createSignal } from "solid-js";

export function Test(show) {
    const x = createSignal(7)[0];
    return <div>
        <Show when={show}
            fallback={<div>
                World
            </div>}>
            <div>
                Hello
            </div>
        </Show>
        <Switch fallback={<p>
                {x()}
                is between 5 and 10
            </p>}>
            <Match when={x() > 10}>
                <p>
                    {x()}
                    is greater than 10
                </p>
            </Match>
            <Match when={x() < 5}>
                <p>
                    {x()}
                    is less than 5
                </p>
            </Match>
        </Switch>
    </div>;
}

